cody-nathan/server.js

```diff
@@ -0,0 +1,194 @@
+  // Put your response here...This corresponds to 3 and 4 and 5 in the diagram. The method in article.js it corresponds to is Article.prototype.updateRecord(). The portion of CRUD that this corresponds to is UPDATE.
+  client.query(
+    `UPDATE articles
+    SET
+      title=$1, author=$2, "authorUrl"=$3, category=$4, "publishedOn"=$5, body=$6
+    WHERE article_id=$7;
+    `,
+    [
+      request.body.title,
+      request.body.author,
+      request.body.authorUrl,
+      request.body.category,
+      request.body.publishedOn,
+      request.body.body,
+      request.params.id
+    ]
+  )
+    .then(function() {
+      response.send('update complete')
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.delete('/articles/:id', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // Put your response here... 3, 4, and 5, as above. The method it works with is Article.prototype.deleteRecord(). This CRUD is DELETE CRUD.
+  client.query(
+    `DELETE FROM articles WHERE article_id=$1;`,
+    [request.params.id]
+  )
+    .then(function() {
+      response.send('Delete complete')
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.delete('/articles', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // Put your response here... 3, 4 and 5 from the diagram. The method it works with is Article.truncateTable(). This CRUD is DELETE CRUD flavor.
+  client.query(
+    'DELETE FROM articles;'
+  )
+    .then(function() {
+      response.send('Delete complete')
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+// COMMENT: What is this function invocation doing?
+// Put your response here... This creates the articles table if it doesn't already exist.
+loadDB();
+
+app.listen(PORT, function() {
+  console.log(`Server started on port ${PORT}!`);
+});
+
+
+//////// ** DATABASE LOADER ** ////////
+////////////////////////////////////////
+function loadArticles() {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // Put your response here... This uses 3 and 4 because it's a query that receives a result. It does not interact directly with any methods from article.js. This CRUD is UPDATE CRUD.
```

Comment:
Technically, this isn't `CRUD` in the sense of a `CRUD route`. But you can define it as one of the four methods, not the `Update` method. Take another look at the function.